### PR TITLE
Added settings.skipRender

### PR DESCRIPTION
**Pull request by pdkn**
new settings.skipRender prop to allow nexrender to execute jsx scripts on .aep files without rendering an output. Useful for interrigation projects for things like default text values that can be stored and then manipulated manually for later rendering.

### CODE REVIEW

**Comment by inlife** [comment]
hey @pdkn 
thank you for the pull request
could you also add similar flag to the `cli`, so that it could be used similarly from the binary form: `--skip-render`, or something along that line

**Comment by pdkn** [comment]
Thanks @inlife, good call - cli is updated

**Comment by inlife** [comment]
great!

**Comment by FRANK-AND-FREE** [comment]
HI, the idea of "skipRender" is great. Especially since the new MediaEncoder now (May update 2020) supports hardware accelerated video encoding to h264 based mp4 files. It's a lot faster than aerender now. I'd rather use a watch folder for rendering now.

However, could it be made possible move the "skipRender" setting to the job description?
I neither use the CLI nor the JavaScript based programmatic approach, but rather just send jobs over to the nexrender-server via the REST API from a Java programm. In that scenario I can't set the skipRender option anywhere, as I understand.

Best regards,
Frank

**Comment by FRANK-AND-FREE** [comment]
Alternatively the setting could be added to the nexrender-worker as a startup parameter?

**Comment by pdkn** [comment]
@FRANK-AND-FREE you can set it in job settings description `settings.skipRender = true`

**Comment by FRANK-AND-FREE** [comment]
Hi @pdkn ,
currently I'm not calling the render(job, settings) function, since I'm not using the JavaScript way of nexrender. I'm requesting renders by sending jobs to the nexrender-server via HTTP request (using the REST API).

**Comment by inlife** [comment]
I think adding `skipRender` support for the worker as well would be the best approach here. Since it's is more platform-specific option rather than job-specific.

**Comment by inlife** [comment]
@pdkn would you like to create a pull request for the worker package as well with this improvement?

**Comment by inlife** [comment]
Also, @FRANK-AND-FREE, could you share with us how would you use MediaEncoder in combo with nexrender? It is a use-case I always wanted to support, would be very interested to get more info/details.

Cheers!

**Comment by FRANK-AND-FREE** [comment]
Hi @inlife,

right now I have a Java based server application that fetches user created project data from a Firebase. The server application than takes the project data, transforms it into a nexrender job and posts it via HTTP to the nexrender job server. My job contains a custom ExtendScript file that nexrender will apply to an After Effects template.
I guess my Java application will try to register when a new output folder containing the After Effects project file with script applied by nexrender is available and then copy it over to a flat watch folder for Media Encoder (since Media Encoder doesn't allow for watching subfolders in the nexrender worker folder automatically). My Java application will then try to register when a finished video is availble and upload that to Firestore for the user to access it.

Long story short: I would pretty much try to handle the necessary file operations myself via a Java server application after Nexrender has applied the custom ExtendScript file to my After Effects template.
Neither do I have an idea how well this approach would scale with many nexrender jobs, nor if I come up with better solutions when being hands on ;-). This is just my very first naive idea of how things could work together.

**Comment by FRANK-AND-FREE** [comment]
As I think of it, all that's missing for the Media Encoder watch folder to basically work is a flat file structure. Maybe nexrender could provide the option to put job files not in their own subfolder but in the working directory directly (uniquely named)?

**Comment by pdkn** [comment]
I'll need to look at the server logic as haven't got as far as network rendering yet. It's on the list. However I would argue that this isn't either job or platform specific but context specific. i.e I added it as wanted to do do a pass on the .aep templates to run some custom jsx that outputs default values of dynamic content as json. and then do another pass to do the actual render with updated dynamic content. Which would be the same approach when working in the cloud. Therefore, without looking at the server code yet, I'd suggest being able to pass the settings option to it? If that is feasible?

**Comment by pdkn** [comment]
@inlife I'm on a different project at mo so it might take a few weeks to get to this now - just to set expectations in case anyone what's a crack at it themselves in meantime.

**Comment by inlife** [comment]
It seems to me, this issue grows to become a new feature. The general idea which I gathered from your comments: **ability to set render provider for each job**

Render provider can have possible values of: 
* `aerender` - the default renderer, will use existing after effects binary to render the whole job
* `media-encoder` - option to use media encoder instead of aerender to render the job
* `none` - option to skip rendering altogether, just execute scripting, pre and post actions and exit

**Comment by FRANK-AND-FREE** [comment]
Sounds very reasonable. It would be really great to be able to make use of Media Encoder for rendering. It really is a lot faster since the last update.

**Comment by inlife** [comment]
That would be quite a major feature, and would probably go in the `v2.0.0`. Meanwhile I'm just gonna add `--skip-render` option to worker as well, similarly to the cli.

**Comment by FRANK-AND-FREE** [comment]
Fantastic! Thank you so much. I will give it a try soon. :-)
